### PR TITLE
Kai - Fixed a Bug for the main search

### LINKED ISSUE #13: BUG: The Main Course Search Page Access Looks like Part of the Logo
### Have you updated the title?

- [X] I have updated the issue's title.

### Summary

The landing page for Courses takes you to the main search. This is the one where you can actually search for valid courses and add them to your schedule.
Screenshot 2024-05-15 at 5 20 17 PM

However, there are currently three different search pages that all looks about the same, and it may be difficult for the user to find the right one that will allow them to add a course to the schedule once the Course Page gets removed.

### Expected Behavior

The main search page should be easy to reach, either by converting the top left "UCSB Course Search" to a button or adding an unobtrusive button that can lead the user to the right location. Keyword here is unobtrusive.

### Current / Observed Behavior

However, let's say you leave the courses page to look at your schedule, how do you get back to that main page? A user might accidentally go to one of the other searches. To make life easier for the user, what if we just make it clear that this is a button or at least some sort of link that will direct you to the main course search page.
Screenshot 2024-05-15 at 5 22 58 PM

### Steps to Reproduce

1. Go to the page https://courses.dokku-00.cs.ucsb.edu or local
2. Search for a course you would like to add
3. An error will ask you to make a schedule first
4. After you have made a schedule, assuming you didn't read what we said above regarding the logo and try to look for the course search page

### Browser / Platform Info

_No response_

### PR DESCRIPTION
**Pull request by kai-maeda**
Bug: Landing page takes user to the main search. There are many different search pages and the main page may be difficult to find. Currently the user must click on the logo.

Fix: Another button in the navbar called main search has been placed so that the user can easily find the main page. This is in addition to the logo that does the same function.

Before
Screenshot 2024-05-23 at 6 32 55 PM

After
Screenshot 2024-05-23 at 6 32 21 PM

Dokku: https://proj-courses-kai-maeda-dev.dokku-02.cs.ucsb.edu/ 

Closes #13

### CODE REVIEW

**Review by pconrad — DISMISSED** [review]
Please remove the node_modules directory before we review.

See: 
https://ucsb-cs156.github.io/topics/pull_requests/pull_requests_unwanted_files.html

**Review by kmflippo — DISMISSED** [review]
LGTM

**Review by jchanDev — DISMISSED** [review]
LGTM!

File: frontend/src/main/components/Nav/AppNavbar.js [hunk 1]
```diff
@@ -63,6 +63,11 @@ export default function AppNavbar({
                   Personal Schedules
                 </Nav.Link>
               )}
+              {hasRole(currentUser, "ROLE_USER") && (
+                <Nav.Link href="/" data-testid="appnavbar-mainsearch-list">
+                  Main Search
+                </Nav.Link>
+              )}
```
Comment (pconrad) [inline frontend/src/main/components/Nav/AppNavbar.js]: Why are we gating this on ROLE_USER?

That requires folks to be logged in.  I don't think we've ever wanted folks to have to be logged in to do simple searchers for courses.  They only need to be logged in if they are building personal schedules or doing admin features.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "devDependencies": {
+    "prettier": "^3.2.5"
+  }
+}
```
Comment (pconrad) [inline package.json]: Why?

Comment (pconrad) [reply package.json]: Oh, snap... this is not in the frontend directory.  This definitely doesn't belong here.  Please remove it.

**Review by pconrad — DISMISSED** [review]
I have two questions....

(1) I wonder why we are putting ROLE_USER on the basic search? We've never done that before, and I don't think we want to.  Do you see it differently?

(2) Why did you add a package.json at the top level containing prettier?  We typically only work with npm inside the frontend directory.  I think prettier is already there.

**Review by kmflippo — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**